hybrid_identity.py

```diff
@@ -60,7 +60,8 @@ def authenticate(self, user_id, password):
             assert utils.check_password(password, user_ref['password'])
         except TypeError:
             raise AssertionError('Invalid user / password')
-        except KeyError:  # if it doesn't have a password, it must be LDAP
+        except (ValueError, KeyError):
+            # if it doesn't have a password, it must be LDAP
```

Comment:
The error I get without catching ValueError is. If you have any insight on the error, I would appreciate any information.

2015-01-16 18:06:57.958 20417 INFO passlib.registry [-] registered crypt handler 'sha512_crypt': <class 'passlib.handlers.sha2_crypt.sha512_crypt'>
2015-01-16 18:06:58.011 20417 ERROR keystone.common.wsgi [-] invalid sha512-crypt hash
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi Traceback (most recent call last):
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/common/wsgi.py", line 223, in __call__
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     result = method(context, **params)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/token/controllers.py", line 100, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     context, auth)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/token/controllers.py", line 296, in _authenticate_local
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     password=password)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/notifications.py", line 307, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     result = f(wrapped_self, context, user_id, *args, **kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 202, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return f(self, *args, **kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 213, in wrapper
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return f(self, *args, **kwargs)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/core.py", line 549, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     ref = driver.authenticate(entity_id, password)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/identity/backends/hybrid_identity.py", line 63, in authenticate
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     assert utils.check_password(password, user_ref['password'])
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/keystone/common/utils.py", line 139, in check_password
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     return passlib.hash.sha512_crypt.verify(password_utf8, hashed)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/passlib/utils/handlers.py", line 464, in verify
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     self = cls.from_string(hash)
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi   File "/usr/lib/python2.6/site-packages/passlib/handlers/sha2_crypt.py", line 454, in from_string
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi     raise ValueError("invalid sha512-crypt hash")
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi ValueError: invalid sha512-crypt hash
2015-01-16 18:06:58.011 20417 TRACE keystone.common.wsgi 
2015-01-16 18:06:58.038 20417 INFO eventlet.wsgi.server [-] 10.90.1.10 - - [16/Jan/2015 18:06:58] "POST /v2.0/tokens HTTP/1.1" 500 398 2.412308